extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts

```diff
@@ -576,6 +580,28 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
             this.config.autoExceptionInstrumented = true;
         }
 
+        if (this.config.disableExceptionTracking === false &&
+            !this.config.autoUnhandledPromiseInstrumented) {
+            // We want to enable exception auto collection and it has not been done so yet
+            const onunhandledrejection = "onunhandledrejection";
+            const originalOnUnhandledRejection = window[onunhandledrejection];
```

Comment:
As part of my unload PR, I've added a getWindow() helper method which you will need to use as the SDK can be run in an environment without window, document, navigator etc defined. And this causes a ReferenceError exception to be thrown when accessing window in one of these environments.

Hopefully, the react issue and my PR can be checked in shortly so we can pick up your change as well.

Reply:
My Unload and EnvUtils (hasWindow(), getWindow()) is now checked in and I've disabled the failing reactivenative tests

Reply:
Shall I merge in that branch and use the `getWindow` function?

Reply:
If you rebase to master you will get the methods and then yes, if you use getWindow() (Checking that it doesn't return null and/or use hasWindow() before hooking the events) then this looks good.

Reply:
Updated the PR with a rebase of `master` and using the new method (see https://github.com/microsoft/ApplicationInsights-JS/pull/1070/commits/4788af78d584c90087ab1d6ad910702af2b2d344)